### PR TITLE
add release pipeline: builds, verifies BITCOIN_DA_ID, releases binary

### PR DESCRIPTION
**Pull request by cetin**
# Description

add release pipeline: builds, verifies BITCOIN_DA_ID, releases citrea binary

## Linked Issues
- Fixes # (issue, if applicable)
- Related to # (issue) 

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

**Review by Chinacolt — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RUN apt update && apt -y install curl gcc cpp cmake clang llvm && apt -y autoremove && apt clean && rm -rf /var/lib/apt/lists/*
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+RUN /root/.cargo/bin/cargo install cargo-binstall
+RUN /root/.cargo/bin/cargo binstall -y cargo-risczero
+RUN /root/.cargo/bin/cargo risczero install --version v2024-04-22.0
+
+WORKDIR /citrea
+COPY . .
+
+RUN /root/.cargo/bin/cargo build --release
```
Comment (eyusufatik) [inline Dockerfile:12]: use env var REPR_GUEST_BUILD=1 here

e.g. 

```sh
REPR_GUEST_BUILD=1 cargo build --release
```

**Review by eyusufatik — DISMISSED** [review]

File: Dockerfile
[the remark this answers is not in the document]
Comment (Chinacolt) [reply Dockerfile:1]: It is irrelevant to auto-release, and we no longer use a Docker file to build Citrea. Please remove this file from this PR.

File: .github/workflows/release.yml
Comment (Chinacolt) [inline .github/workflows/release.yml:1]: This seems ok to me. If this worked on the DevOps repo, we can continue.

**Review by Chinacolt — CHANGES_REQUESTED** [review]
release.yml seems ok to me. If this worked on the DevOps repo, we can continue.

Dockerfile is is irrelevant to auto-release, and we no longer use a Docker file to build Citrea. Please remove this file from this PR.

File: Dockerfile (hunk 1 above)
Comment (Chinacolt) [reply Dockerfile:12]: This is added to GitHub actions Yaml directly, so it is resolved.

**Review by Chinacolt — APPROVED** [review]

**Pull request merged**